bundles/com.espressif.idf.core/src/com/espressif/idf/core/variable/OpenocdVariableResolver.java

```diff
@@ -0,0 +1,76 @@
+package com.espressif.idf.core.variable;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.variables.IDynamicVariable;
+import org.eclipse.core.variables.IDynamicVariableResolver;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.embedcdt.core.EclipseUtils;
+import org.eclipse.launchbar.core.ILaunchBarManager;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.IDFEnvironmentVariables;
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.core.util.StringUtil;
+
+public class OpenocdVariableResolver implements IDynamicVariableResolver
+{
+	private static final String OPENOCD_PREFIX = "com.espressif.idf.debug.gdbjtag.openocd"; //$NON-NLS-1$
+	private static final String INSTALL_FOLDER = "install.folder"; //$NON-NLS-1$
+
+	public String resolveValue(IDynamicVariable variable, String argument) throws CoreException
+	{
+		return getAppropriateEnumVariable(variable).map(this::resolveForOpenocdDynamicEnum).orElse(variable.getName());
+	}
+
+	private Optional<OpenocdDynamicVariable> getAppropriateEnumVariable(IDynamicVariable variable)
+	{
+		return Arrays.stream(OpenocdDynamicVariable.values()).filter(v -> v.getValue().equals(variable.getName()))
+				.findAny();
```

Comment:
added this to make code more robust, consider this use case:
- adding the variable to the extension point, but not adding it to the openocd enum. 
- In this case, returning striped value without resolving it;